core/src/main/java/com/linecorp/armeria/internal/common/RequestContextUtil.java

```diff
@@ -48,6 +58,57 @@
     private static final Set<Thread> REPORTED_THREADS =
             Collections.newSetFromMap(new MapMaker().weakKeys().makeMap());
 
+    /**
+     * The default {@link ContextStorage} which stores the {@link RequestContext} in {@link FastThreadLocal}.
+     */
+    public static final ContextStorage defaultContextStorage = new ThreadLocalContextStorage();
+
+    private static final ContextStorage contextStorage;
+
+    static {
+        final List<ContextStorageProvider> providers = ImmutableList.copyOf(
+                ServiceLoader.load(ContextStorageProvider.class));
+        final String contextStorageFqcn = Flags.contextStorage();
+        if (!providers.isEmpty()) {
+            if (providers.size() > 1) {
```

Comment:
Ah I was thinking instead of loading directly from the flag, we'd use the flag to disambiguate among the different providers in this case. I guess we can do both though

Reply:
Ah, I guess I totally misunderstood. 🤣
So you meant when multiple providers exist (e.g `com.x.AProvider`, `io.x.BProvider`...), if the flag is `io.x.BProvider`, then we should choose the `BProvider` class. Right?

If so, I want to get rid of the current way and follow it because, I think, there would be a case who wants to use two different JARs that have their own `RequestContextStorage`.
So I think we should choose one of them instead of throwing an exception. 😉